spec/did_you_know_spec.rb

```diff
@@ -0,0 +1,68 @@
+RSpec.describe Geordi::Hint do
+
+  subject { Geordi::Hint }
+
+  describe '#did_you_know' do
+
+    it 'prints a random hint in the console if an array with suitable suggestions was provided and returns all generated hints' do
+      allow(Geordi::Hint).to receive(:hint).with(:branch).and_return('Did you know? `geordi branch` can check out a feature branch based on a story from Pivotal Tracker')
+      allow(Geordi::Hint).to receive(:hint).with([:branch, :from_master]).and_return('Did you know? `geordi branch -m` can branch from master instead of the current branch')
+      allow(Geordi::Hint).to receive(:hint).with('`geordi branch`').and_return('Did you know? `geordi branch`')
+
+      array = [
+        :branch,
+        [:branch, :from_master],
+        '`geordi branch`'
+      ]
+      expect {subject.did_you_know(array)}.to output(/Did you know\? `geordi branch/).to_stdout
+      expect(subject.did_you_know(array)).to eq([
+        'Did you know? `geordi branch` can check out a feature branch based on a story from Pivotal Tracker',
+        'Did you know? `geordi branch -m` can branch from master instead of the current branch',
+        'Did you know? `geordi branch`'
+      ])
+    end
+
+    it 'prints nothing to the console if an empty array was provided' do
+      array = []
+      expect {subject.did_you_know(array)}.not_to output(/.+/).to_stdout
+    end
+
+    it 'prints nothing to the console if an array with nil elements was provided' do
+      array = [nil, nil]
+      expect {subject.did_you_know(array)}.not_to output(/.+/).to_stdout
+    end
```

Comment:
Calling `Hint.did_you_know` with either no arguments or `nil` values should not silently proceed, but both crash in the face of the developer. It is not intended behavior, but misusage. In this light, you _might_ add a test that expects the exception, but I'd leave this out altogether.